iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/TieLineImpl.java

```diff
@@ -128,9 +159,15 @@ public double getB2() {
 
         @Override
         public HalfLineImpl setB2(double b2) {
+            double oldValue = this.b2;
             this.b2 = b2;
+            notifyUpdate("b2", oldValue, b2);
             return this;
         }
+
+        private String getHalfLineAttribute() {
+            return this == parent.half1 ? "1" : "2";
```

Comment:
Maybe you should return "half1" or "half2" instead of "1" and "2", to avoid to many concatenations